ctags.go

```diff
@@ -30,7 +30,7 @@ type Entry struct {
 }
 
 type Parser interface {
-	Parse(path string, content []byte) ([]*Entry, error)
+	Parse(path string, content []byte) ([]*Entry, *ParseError)
```

Comment:
I'm looking for feedback on this from our Go experts. Is it weird to have a custom error type in a library API? And is it weird to use a pointer (I liked it because it can easily be used as an `error`).

Reply:
(landed on this PR on a mislick in my notifications, but here I am 😅 )

Returning a concrete error implementation is a huge gotcha: https://github.com/sourcegraph/conc/pull/75 - I think general practice is to return `error`, and document on interface/package that the caller can/should cast this into `*ParsedError`, like `net.DNSError` and friends https://pkg.go.dev/net#DNSError

Reply:
Wow, I didn't know about that gotcha. I agree with @bobheadxi to let the clients switch on the underlying error. 

We should export key error types which communicate to users what happened and let the caller decide what is fatal or not.

Reply:
yes I would return error here. Then at call sites you need to do something like:
```go
entries, err := parser.Parse(path, content)
if err != nil {
  var parseError *ctags.ParseError
  if errors.As(err, &parseError) {
    // do parseError specific behaviour ...
  } else {
    // ...
  }
}
```

Reply:
Thanks, this is really helpful! Will fix.